### PR TITLE
Set IDE window maximized at initialization step

### PR DESCRIPTION
**Pull request by vzhukovs**
Set `IDEA_WINDOW` window type maximized at initialization step. Patch fix for https://youtrack.jetbrains.com/issue/PRJ-452
Related discussion: https://youtrack.jetbrains.com/issue/PRJ-452#focus=Comments-27-4861019.0-0

fixes: https://github.com/eclipse/che/issues/19595

Now after workspace create, the IDE window is maximized:

Eclipse Che 2021-04-30 13-47-18

Signed-off-by: Vladyslav Zhukovskyi <vzhukovs@redhat.com>

### CODE REVIEW

**Comment by vzhukovs** [comment]
Devfile to test:

```yaml
apiVersion: 1.0.0
metadata:
  name: che-idear-vzhukovs-quay
components:
  - type: cheEditor
    reference: 'https://gist.github.com/vzhukovs/4bf9fb9b932031dae5cbfa912292b281/raw/eee8b4b151ddc72c483d90caf76d761dcce2a834/che-idea-experiments-meta.yaml'
    alias: che-idea
```

**Comment by benoitf** [comment]
hello, do you think that we should wait to have this patch merged upstream first, to avoid to have a bigger set of patches on our own ?

File: patches/projector-client/PRJ-452.patch [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ import org.jetbrains.projector.common.protocol.toClient.WindowType
++import org.jetbrains.projector.common.protocol.toServer.ClientResizeEvent
+ import org.jetbrains.projector.common.protocol.toServer.ClientWindowCloseEvent
+ import org.jetbrains.projector.common.protocol.toServer.ClientWindowMoveEvent
+ import org.jetbrains.projector.common.protocol.toServer.ClientWindowResizeEvent
+@@ -118,6 +121,7 @@ class Window(windowData: WindowData, private val stateMachine: ClientStateMachin
+ 
+     if (windowData.windowType == WindowType.IDEA_WINDOW || windowData.windowType == WindowType.POPUP) {
+       canvas.style.border = "none"
++      setMaximized()
```
Comment (benoitf) [inline patches/projector-client/PRJ-452.patch:30]: does it means that even popup are maximized ?

**Review by azatsarynnyy — APPROVED** [review]
I tested it with provided Devfile. The main IDEA window is maximized, by default.
Also, I've checked resizing when using the dashboard's "hamburger" button. Everything works correctly. Cool!
@vzhukovs do we have a corresponding upstream patch?

**Comment by vzhukovs** [comment]
>hello, do you think that we should wait to have this patch merged upstream first, to avoid to have a bigger set of patches on our own ?

Taking into account that fact, that on projector-client not many people are working, and how long upstream PRs are merging: https://github.com/JetBrains/projector-client/pull/19 I think, we can fix it on our side and propose the fix into upstream. Anyway, when the upstream issue closes, it closes with the particular fix in upstream repo. Then patch from downstream repo removes and sticked version for both [projector-client](https://github.com/che-incubator/jetbrains-editor-images/blob/master/clone-projector.sh#L36) and [projector-server](https://github.com/che-incubator/jetbrains-editor-images/blob/master/clone-projector.sh#L49) updates.

File: patches/projector-client/PRJ-452.patch (hunk 1 above)
Comment (vzhukovs) [reply patches/projector-client/PRJ-452.patch:30]: Probably, I need to update the method name to clarify more about it. The main action is performed to the popup and IDEA window type is to set the client size where should projector-server draw the windows, some kind of viewport. Agree, it misleading a bit. It doesn't perform the actual IDEs resize or maximize, but tells the projector-server about maximizing the area in the browser document where the UI should be rendered by using `window.inner[Width, Height]/window.devicePixelRatio`.

**Comment by vzhukovs** [comment]
>do we have a corresponding upstream patch?

Not yet. I'm checking how it works on electron version at this moment.

**Comment by azatsarynnyy** [comment]
@vzhukovs please merge if it's done

**Pull request merged**